src/ui/components/Button/index.js

```diff
@@ -24,12 +37,22 @@ export default class Button extends React.Component {
       className,
       href,
       to,
+      type,
+      width,
       ...rest
     } = this.props;
     const props = { ...rest };
 
+    if (!type || !BUTTON_TYPES.includes(type)) {
+      throw new Error(`"${type}" supplied but that is not a valid button type`);
+    }
+
     const setClassName = (...classConfig) => {
-      return classNames('Button', className, ...classConfig);
+      return classNames(
+        'Button', `Button--${type}`, className, ...classConfig, {
+          width: `Button--width-${width}`,
```

Comment:
I dont really like this freestyle here. I think it would be nice to have a `boolean` prop called `fullwidth` (default: `false`) and if set, then we add the `Button--widthfull` or `Button-fullwidth` class.